moto/swf/responses.py

```diff
@@ -0,0 +1,311 @@
+import json
+import six
+
+from moto.core.responses import BaseResponse
+from werkzeug.exceptions import HTTPException
+from moto.core.utils import camelcase_to_underscores, method_names_from_class
+
+from .models import swf_backends
+
+
+class SWFResponse(BaseResponse):
+
+    @property
+    def swf_backend(self):
+        return swf_backends[self.region]
+
+    # SWF actions are not dispatched via URLs but via a specific header called
+    # "x-amz-target", in the form of com.amazonaws.swf.service.model.SimpleWorkflowService.<action>
+    # This is not supported directly in BaseResponse sor for now we override
+    # the call_action() method
+    # See: http://docs.aws.amazon.com/amazonswf/latest/developerguide/UsingJSON-swf.html
+    def call_action(self):
+        headers = self.response_headers
+        # Headers are case-insensitive. Probably a better way to do this.
+        match = self.headers.get('x-amz-target') or self.headers.get('X-Amz-Target')
+        if match:
+            # TODO: see if we can call "[-1]" in BaseResponse, which would
```

Comment:
Awesome! I removed this accordingly, and I rebased the whole PR against master. I'll `git push -f` now, hope you don't mind (my current pythonistas team seem to favor that for topic branches, don't know if it's a general practice in python projects).